packages/logger/src/pipe.ts

```diff
@@ -38,14 +38,18 @@ export enum PipeMode {
 function pipeSpawnOutput(
   logger: bunyan,
   { stdout, stderr }: SpawnOutput = {},
-  { mode = PipeMode.COMBINED, lineTransformer }: PipeOptions = {}
+  { mode = PipeMode.COMBINED, lineTransformer }: PipeOptions = {},
```

Comment:
If adding, why not add it to `PipeOptions`? Maybe then https://github.com/expo/eas-build/pull/299 wouldn't even be necessary?

Reply:
Would still need to update `SpawnOptions` in `turtle-spawn/src/index.ts` if I'm not mistaken

Reply:
👍 Wouldn't it be cooler to export `PipeOptions` and `SpawnOutput` from `logger` and import (& re-export) in `turtle-spawn`?

Reply:
Hm, I think I'm tired, but I don't see what problem would that solve. Can you elaborate on that?

Reply:
`SpawnOptions` defined manually in `turtle-spawn` https://github.com/expo/eas-build/blob/01defae0efe516a2378bf8375a11842457e8b05f/packages/turtle-spawn/src/index.ts#L8-L12 merge `SpawnAsyncOptions` (that is ok to me) and manually: `logger`, `lineTransformer` and `mode`. The `logger` is indeed used by the `spawn`, but `lineTransformer` and `mode` are closely (and only) related to `pipeSpawnOutput`.

Unless I don't see something, there's no reason to define `lineTransformer` and `mode` manually in `turtle-spawn`. Instead we should export `PipeOptions` and define `SpawnOptions = SpawnAsyncOptions & PipeOptions & { logger }`. This way no types need to be changed if we add/remove options from either of the packages?

Reply:
Ohhh, right. Yeah that does sound reasonable